rtengine/imagedata.cc

```diff
@@ -320,7 +320,9 @@ FramesData::FramesData(const Glib::ustring &fname, time_t ts) :
             //orientation = pos->print(&exif);
         }
 
-        if (find_exif_tag("Exif.NikonLd4.LensIDNumber") || find_exif_tag("Exif.Sony2.LensID")) {
+        if ((find_exif_tag("Exif.NikonLd4.LensID") && pos->toLong()) ||
```

Comment:
The reason I placed LensID first is because the correct lens name was not read when I placed it after.

Reply:
That's a bit strange... Mind sharing sample?

Reply:
With Exiv2 0.27.5, the Z 6_2 image provided by @jade-nl in https://github.com/Beep6581/RawTherapee/issues/6291#issuecomment-869113552 shows up as "VR 105mm f/2.8G" (from Exif.Photo.LensModel) and Exif.Photo.LensMake is blank. Though Lensfun automatic detection works, the name is not precise an may cause issues in the future. The ExifTool-based rtexif is using LensModel and LensMake only if the lens name is not found elsewhere (for example, the LensID as interpreted by ExifTool).

Based on the above observations, I feel the best approach is to first check if Exiv2 has an interpretation of the LensID (i.e. the printed string of LensID is not equal to the LensID integer as a (base10?) string), then falling back to LensModel. You seem to have more knowledge of Exiv2, so what do you think of this approach?

Reply:
Ah, but that was because it was an F-mount lens w/ FTZ adapter. Please read the logic I proposed again - prioritize translated NikonLd4.LensIDNumber (if LensID is 0), otherwise (new Z lens on Z body) Photo.LensModel is usable.

Reply:
Ok, I did not notice that comment, but how do we know LensModel is correct for Z lenses? What about older F-mount cameras? Do they not use LensModel and therefore it's safe to use Exif2::lensName, or must we use Exif.NikonLd3.LensIDNumber and others?

Reply:
In the meantime, I changed the logic to use use Exiv2::lensName if Exiv2 cannot interpret LensID or LensIDNumber.

Reply:
I guess that could work as well.